### PR TITLE
fix(ddtrace/tracer): fix UDS handling in WithAgentURL (#3553)

### LINKED ISSUE #3553: [BUG]: Invalid handling of unix:/... URLs in WithAgentURL
### Tracer Version(s)

v2.0.0-rc.20

### Go Version(s)

go version go1.24.3 linux/amd64

### Bug Report

Unix URLs cannot be handled like this:
```go
		switch u.Scheme {
		case "unix", "http", "https":
			c.agentURL = &url.URL{
				Scheme: u.Scheme,
				Host:   u.Host,
			}
		default:
			log.Warn("Unsupported protocol %q in Agent URL %q. Must be one of: http, https, unix.", u.Scheme, agentURL)
		}
```

`u.Host` is always empty, `u.Path` contains the path to the socket.

### Reproduction Code

```go
	traceProvider := ddotel.NewTracerProvider(
		tracer.WithAgentURL("unix:///var/run/datadog/apm.socket"),
		tracer.WithSamplerRate(0.01),
	)
```

### Error Logs

```
WARN: DIAGNOSTICS Unable to reach agent intake: Post "http://UDS_/v0.4/traces": dial unix: missing address
```

### Go Env Output

_No response_

### PR DESCRIPTION
**Pull request by darccio**
### What does this PR do?

Changes how `WithAgentURL` handles Unix Data Sockets paths. It creates a helper function and tags some places where it could be used.

### Motivation

Fix #3553

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] New code is free of linting errors. You can check this by running `golangci-lint run` locally.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: internal/uds.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+package internal
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+func UDSURL(path string) *url.URL {
```
Comment (kakkoyun) [inline internal/uds.go:14]: My eyes hurt 😭

Comment (mtoffl01) [reply internal/uds.go:14]: What about UnixURL?

**Review by nsrip-dd — APPROVED** [review]

**Review by mtoffl01 — APPROVED** [review]
LGTM, some suggestions -

1. UnixURL may be a good alternative name to UDSURL
2. Although unlikely in the real world, it might be good to add a test case when path is nil?
3. IMO you might as well migrate all the `// TODO(darccio): use internal.UDSURL instead` to actually use the new helper function rather than leaving it as a TODO. I would understand if this is just a safety measure to enable v2 release, though

**Comment by darccio** [comment]
I agree with the naming suggestion, I wasn't inspired at the moment. Just a few comments:

> Although unlikely in the real world, it might be good to add a test case when path is nil?

That shouldn't be possible, `path string` wouldn't allow a `nil` value. ITOH, there is a test for blank value.

> IMO you might as well migrate all the // TODO(darccio): use internal.UDSURL instead to actually use the new helper function rather than leaving it as a TODO. I would understand if this is just a safety measure to enable v2 release, though.

It is a safety measure to avoid introducing side effects for v2 release.
